client/httpClient.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"net/http"
+)
+
+var client *http.Client
+
+// Permission encapsulates the tls configurations.
+type Permission struct {
+	CertFile string
+	KeyFile  string
+	CaCert   string
+}
+
+// GetCaCert returns the CertFile if CaCert is empty.
+func (perm *Permission) GetCaCert() string {
+	if perm.CaCert == "" {
+		return perm.CertFile
+	}
+	return perm.CaCert
+}
+
+// SendRequest encapsulates the entire process of initializing the client
+// and sending a request, returning the byte body of the response.
+func SendRequest(permission Permission, request *http.Request) (*[]byte, error) {
+
+	var err error
+
+	if client == nil {
```

Comment:
If I were designing this API, I would instead pass the client. In your case, the client needs to be package-global and permission is only provided to initialize it - so why not initialize the client at the caller's and pass it around instead.
There's no benefit in re-using the client at this point as far as I can see.